### PR TITLE
Fix file watching events being lost

### PR DESCRIPTION
**Pull request by sgraband**
#### What it does

Fixed an issue with tracking the current `FileSystemEvents` in the `MainFileSystemEventService`.
Before, the events were tracked in a global field which was intended to be cleaned after an `onDidFilesChanges()` update was sent to the proxy.
This is problematic as we do not await sending the update.
Thus, the global field will be cleaned before the update is even sent and therefore, file watching updates might get lost.
This change ensures that a field of events is kept for each update.

Note that simply awaiting the update to be sent is not enough.
This would introduce race conditions as other parallel updates might clean the state of other updates before they are being sent.

Fixes # 12260. (Will remove the whitespace and this on the real PR)

Contributed on behalf of STMicroelectronics
#### How to test

1. Install the following extension: [file-watching-sample-0.0.1.zip](https://github.com/eclipse-theia/theia/files/10898166/file-watching-sample-0.0.1.zip)
2. Change/Add/Delete a file
3. All actions should be detected by the extension (should display a console log and a information message in Theia)
#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Comment by tsmaeder** [comment]
It might make it easier for reviewers to explain in what scenario the reassignment leads to an error, like 
1. a happens
2. then b
3. ...
4. Now the array get reassigned
5. Boom!

You know this @sgraband but reviewers have to figure it out.

**Comment by sgraband** [comment]
Thanks for the feedback @tsmaeder. I updated the commit and PR description. Hope it is clearer now!

File: packages/plugin-ext/src/main/browser/main-file-system-event-service.ts [hunk 1]
```diff
@@ -37,13 +37,13 @@ export class MainFileSystemEventService {
...
         const fileService = container.get(FileService);
 
-        // file system events - (changes the editor and other make)
-        const events: FileSystemEvents = {
-            created: [],
-            changed: [],
-            deleted: []
-        };
         this.toDispose.push(fileService.onDidFilesChange(event => {
+            // file system events - (changes the editor and other make)
```
Comment (cdamus) [inline packages/plugin-ext/src/main/browser/main-file-system-event-service.ts:41]:
```suggestion
            // file system events - (changes the editor and others make)
```

**Review by cdamus — APPROVED** [review]
LGTM! Just a typo in a pre-existing comment but that is now owned by this PR because it was moved 😀

**Comment by rschnekenbu** [comment]
close PR as it was merged in theia already (PR 12264)

**Pull request closed without merging**